### PR TITLE
Build and publish initial image

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Review by dmcdii — APPROVED** [review]
lgtm

File: .github/workflows/docker-publish.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+      # Extract metadata (tags, labels) for Docker
+      # https://github.com/docker/metadata-action
+      - name: Extract Docker metadata
+        id: meta
+        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          tags: |
+            type=match,pattern=v(\d+.\d+),group=1
```
Comment (b-dean) [inline .github/workflows/docker-publish.yml]: @krmichelos, what type of versions are you intending to put on this? this seems to be copied from [ontariosystems/docker-jenkins](https://github.com/ontariosystems/docker-jenkins) where the versions match Jenkins versions which only have two parts. If you're intending to have standard semver, you should add another `\.\d+` to the pattern

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,42 @@
+FROM alpine:3.16
+
+# set some defaults
+ENV AWS_DEFAULT_REGION "us-east-1"
+ENV KUBECTL_VER=v0.23.10
+# https://github.com/sgerrand/alpine-pkg-glibc/releases
+ENV GLIBC_VER=2.33-r0
```
Comment (b-dean) [inline Dockerfile]: I don't think you should install glibc. Alpine can behave oddly when you don't just use musl. 

We do that here:
https://bitbucket.org/ontsys/dind-rootless-awscli/src/5e0495bddfb7cff7eb507cace3643986baa4b7b1/Dockerfile#lines-3:30,47:51

**Review by b-dean — CHANGES_REQUESTED** [review]
I think we should build awscli using musl instead of glibc. 

Also probably use standard semver tags `/^v\d+\.\d+\.\d+$/`  in the github actions.

File: .github/workflows/docker-publish.yml (hunk 1 above)
Comment (ghost) [reply .github/workflows/docker-publish.yml]: We don't need major versions since we will never break the API ;)

**Review by b-dean — APPROVED** [review]
seems good now

**Pull request merged**